### PR TITLE
Stacks 2.1 updates for Clarinet, Stacks JS and API docs

### PR DESCRIPTION
**Pull request by janniks**
Stacks 2.1 Docs Update

`Visit Preview` via Vercel below to view the latest live site!

### CODE REVIEW

File: docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- [`delegate-stack-extend`](https://github.com/stacksgov/sips/blob/4ed252c11ade5569ab11a28d373b267f0e5499d4/sips/sip-015/sip-015-network-upgrade.md#new-method-delegate-stack-extend) = `client.delegateStackExtend()`
+- [`delegate-stack-increase`](https://github.com/stacksgov/sips/blob/4ed252c11ade5569ab11a28d373b267f0e5499d4/sips/sip-015/sip-015-network-upgrade.md#new-method-delegate-stack-increase) = `client.delegateStackIncrease()`
+
+### New Helper Methods
+
+<!-- todo: add links to api reference, once live -->
+
+- `StackingClient.getAccountBalanceLocked()` gets the locked balance of the address from a node (`/v2` endpoint)
+- `StackingClient.getAccountExtendedBalance()` gets more detailed balances of the address from an API (`/extended` endpoint); this includes STX, FTs, and NFTs
+- `StackingClient.getDelegationStatus()` gets the `get-delegation-info` information of the address from a node (readonly contract call)
```
Comment (trichards-hiro) [inline docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md:50]:
```suggestion
- `StackingClient.getDelegationStatus()` gets the `get-delegation-info` information of the address from a node (read-only contract call)
```

File: docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- `StackingClient.getPoxOperationInfo()` gets information about the current period of transition to PoX-2
+
+## Migration
+
+<!-- todo: version number is not fixated yet, might change -->
+
+Previous `@stacks/stacking` releases should automatically switch to the new PoX contract (in Period 3).
```
Comment (trichards-hiro) [inline docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md:57]:
```suggestion
Previous `@stacks/stacking` releases should automatically switch to the new PoX contract (in Period 3); however, if you want to use the new methods detailed above, you need to update to `@stacks/stacking` to version >=`6.0.0`.
```

File: docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+However, if you want to use the new methods detailed above, you need to update to `@stacks/stacking` to a version >=`6.0.0`.
```
Comment (trichards-hiro) [inline docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md:58]:
```suggestion
```

**Comment by saralab** [comment]
@LakshmiLavanyaKasturi : 
Can we include the updated periods for activation and transition for clarity from this blog: https://www.hiro.so/blog/how-the-stacks-2-1-transition-impacts-stacking ?

File: docs/stacks-2.1-upgrades.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+If you have any trouble with the above command, refer to our [troubleshooting guide](clarinet/troubleshooting.md) or report an issue [here](https://github.com/hirosystems/clarinet/issues).
+
+New stacks-node will spin up. At Bitcoin block height 102, the chainstate will migrate to epoch 2.05, which was a soft introduced earlier this year, reducing the costs of operations, and at bitcoin block height 106, the chainstate will migrate to epoch 2.1.
+
+As shown in the below screenshot, the epoch changes are indicated in the *Transactions* section `deployed: ST00000000000000002AMW42H.costs-3 (ok true).`
+
+![image](../docs/images/integrate-epoch-changes.png)
+
+If the contracts you're developing use Clarity 2, you must wait for this epoch to cross before deploying your contracts.
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:74]: Appears to be an important callout. Perhaps add emphasis or bold or as a NOTE?

File: docs/stacks-2.1-upgrades.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+> **_NOTE:_**
+> This new feature is experimental. If you have trouble or find issues, you can report them [here](https://github.com/hirosystems/clarinet/issues).
+
+#### Limitations
+
+In the current release, contracts are not automatically deployed when running `clarinet integrate` with Stacks 2.1. The default devnet deployment needs to be manually applied using the command:
+
+`$ clarinet deployments apply --devnet`
+
+Clarinet has been updated with new features to work with Stacks 2.1. For more detailed information about these updates, refer to [new features in Clarinet](https://www.hiro.so/blog/meet-4-new-features-in-clarinet).
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:94]: The blog linked here includes new improvements to Clarinet, but aren't entirely Stacks 2.1 specific, though.

File: docs/stacks-2.1-upgrades.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+## Stacks Blockchain API
+
+### Distinguish contract versions
+
+The endpoints ` /extended/v1/tx/<txid>` and `/extended/v1/contract/<contract-principal>` returns a new property, `clarity_version.`
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:100]: We could add a statement explaining what this section is about to unpack. Feels odd to jump right into an endpoint w/o any primer.

File: docs/stacks-2.1-upgrades.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+"clarity_version": 2
+
+A sample response for a regular (non-versioned) smart contract transaction with Clarity version null is shown below:
+
+"clarity_version": null
+
+### Coinbase payouts
+
+The Stacks 2.1 updates support alternate recipients for coinbase transactions. The `alt_recipient` value is passed in the blockchain transaction, and the API returns a new coinbase `alt_recipient` property on applicable requests. ; however, if you want to use the latest methods detailed above, you need to update to `@stacks/stacking` to version >=`6.0.0`.
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:112]:
```suggestion
The Stacks 2.1 updates support alternate recipients for Coinbase transactions. The `alt_recipient` value is passed in the blockchain transaction, and the API returns a new Coinbase `alt_recipient` property on applicable requests; however, if you want to use the latest methods detailed above, you need to update to `@stacks/stacking` to version >=`6.0.0`.
```

File: docs/stacks-2.1-upgrades.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+The `alt_recipient` value varies based on the standard vs. contract principal recipients.
+
+Standard principal recipient:
+`"alt_recipient": "ST2X2FYCY01Y7YR2TGC2Y6661NFF3SMH0NGXPWTV5" <-- New`
+
+Contract principal recipient:
+`"alt_recipient": "ST2X2FYCY01Y7YR2TGC2Y6661NFF3SMH0NGXPWTV5.hello_world" <-- New`
+
+### PoX2 support for Rosetta API
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:122]: We use PoX-2 and PoX2 formats; it'd be better to follow a single format for consistency.

File: docs/stacks-2.1-upgrades.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+
+The PoX-2 support is extended to various functions mentioned [below](#new-pox-interactive-functions-for-clarity) and [Rosetta API](https://www.rosetta-api.org/).
+
+- **Address format changes**: Adding support for segwit/taproot/bech32 addresses in PoX 2.
+- **Constructing stacking operations**: Rosetta has built-in support for performing various "Stacking" operations, which perform PoX-1 contract calls under the hood. These need to be changed to PoX-2 contract calls.
+- **Parsing stacking operations**: Adding support for performing various "Stacking" operations involving PoX-2 contract calls.
+
+## Stacks.js
+
+Mention about three sections below.
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:132]: Confused about "mention" - copy+paste error?

File: docs/stacks-2.1-upgrades.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+- [`stack-increase`](https://github.com/stacksgov/sips/blob/7c6c69d37c0ab46c0c782bbb203f9eea6d4d42a4/sips/sip-015/sip-015-network-upgrade.md#new-method-stack-increase) = `client.stackIncrease()`
+- [`delegate-stack-extend`](https://github.com/stacksgov/sips/blob/7c6c69d37c0ab46c0c782bbb203f9eea6d4d42a4/sips/sip-015/sip-015-network-upgrade.md#new-method-delegate-stack-extend) = `client.delegateStackExtend()`
+- [`delegate-stack-increase`](https://github.com/stacksgov/sips/blob/7c6c69d37c0ab46c0c782bbb203f9eea6d4d42a4/sips/sip-015/sip-015-network-upgrade.md#new-method-delegate-stack-increase) = `client.delegateStackIncrease()`
+- [`stack-aggregation-commit-indexed`](https://github.com/stacksgov/sips/blob/7c6c69d37c0ab46c0c782bbb203f9eea6d4d42a4/sips/sip-015/sip-015-network-upgrade.md#new-method-stack-aggregation-commit-indexed) = `client.stackAggregationCommitIndexed()`
+- [`stack-aggregation-increase`](https://github.com/stacksgov/sips/blob/7c6c69d37c0ab46c0c782bbb203f9eea6d4d42a4/sips/sip-015/sip-015-network-upgrade.md#new-method-stack-aggregation-increase) = `client.stackAggregationIncrease()`
+
+### New helper methods
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:148]: Under each section, let's add a simple description as to what it is about to unpack.

**Review by sabbyanandan — CHANGES_REQUESTED** [review]

File: docs/stacks-2.1-upgrades.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
+---
+title: Migrate Stacking to 2.1
```
Comment (janniks) [inline docs/stacks-2.1-upgrades.md:2]: Title should maybe be something more general. The article is not only about "stacking"

File: docs/stacks-2.1-upgrades.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+
+Now, let's dive into each product and understand the new features or improvements to the existing features for Stacks 2.1 updates.
+
+## Clarinet
+
+### Update Devnet settings
+
+If you are new to Clarinet, refer to the [Getting started](../docs/clarinet/getting-started.md) guide to install clarinet, [set up local development environment](clarinet/how-to-guides/how-to-set-up-local-development-environment.md) and [create your project](clarinet/how-to-guides/how-to-create-new-project.md)
+
+To upgrade your clarinet version, you can refer to [install from a pre-built binary](clarinet/getting-started.md#install-from-a-pre-built-binary) or [install from source using Cargo](clarinet/getting-started.md#install-from-source-using-cargo).
```
[the remark this answers is not in the document]
Comment (LakshmiLavanyaKasturi) [reply docs/stacks-2.1-upgrades.md:49]: I don't think so from my earlier conversations. But double-checking with Brice to make sure I don't miss anything.

File: docs/stacks-2.1-upgrades.md [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+
+## Stacks Blockchain API
+
+The following are the updates for the API endpoints with the Stacks 2.1 updates.
+
+### Distinguish contract versions
+
+This section explains the new property returned with the following existing endpoints.
+
+The endpoints [`/extended/v1/tx/<txid>`](/extended/v1/tx/<txid>) and [`/extended/v1/contract/<contract-id>`](https://docs.hiro.so/api#tag/Smart-Contracts/operation/get_contract_by_id) now returns a new property, `clarity_version.`
```
Comment (janniks) [inline docs/stacks-2.1-upgrades.md:123]: `[`/extended/v1/tx/<txid>`](/extended/v1/tx/<txid>)` this shouldn't be a link or point to something working (like the following link)

File: docs/stacks-2.1-upgrades.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (LakshmiLavanyaKasturi) [reply docs/stacks-2.1-upgrades.md:54]: Brice confirmed that there is no requirement for 2.1 specific upgrade steps

File: docs/stacks-2.1-upgrades.md (hunk 5 above)
Comment (LakshmiLavanyaKasturi) [reply docs/stacks-2.1-upgrades.md:113]: Agreed. Removing this as its not required here.

File: docs/stacks-2.1-upgrades.md (hunk 13 above)
Comment (LakshmiLavanyaKasturi) [reply docs/stacks-2.1-upgrades.md:123]: Ok. Removing the links per this conversation.

File: docs/stacks-2.1-upgrades.md [hunk 14]
```diff
@@ -39,23 +39,23 @@ To understand more about PoX-2 activation, reward cycles, and phases, refer to t
...
 
-This sections helps you update your Devnet settings to work with the Stacks 2.1 updates.
+This sections helps you update your Devnet settings to work with the Stacks 2.1 upgrade.
 
 If you are new to Clarinet, refer to the [Getting started](../docs/clarinet/getting-started.md) guide to install clarinet, [set up local development environment](clarinet/how-to-guides/how-to-set-up-local-development-environment.md) and [create your project](clarinet/how-to-guides/how-to-create-new-project.md)
 
 To upgrade your clarinet version, you can refer to [install from a pre-built binary](clarinet/getting-started.md#install-from-a-pre-built-binary) or [install from source using Cargo](clarinet/getting-started.md#install-from-source-using-cargo).
 
-After installing the latest version of Clarinet, navigate to your project directory and then to `settings/Devnet.toml`.
+After installing the latest version of Clarinet, navigate to your project directory and then to `settings/Devnet.toml` (here after referred to as **devnet.toml**).
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:56]:
```suggestion
After installing the latest version of Clarinet, navigate to your project directory and then to `settings/Devnet.toml` (hereafter referred to as **devnet.toml**).
```

File: docs/stacks-2.1-upgrades.md [hunk 15]
```diff
@@ -39,23 +39,23 @@ To understand more about PoX-2 activation, reward cycles, and phases, refer to t
 
 ![image](../docs/images/stacks_2.1_periods.jpeg)
 
-Now that you have a general understanding of Stacks 2.1 and the PoX changes and what to expect let's dive into each product and understand the new features or improvements to the existing features for Stacks 2.1 updates.
+Now that you have a general understanding of Stacks 2.1 and the PoX changes and what to expect let's dive into each product and understand the new features or improvements to the existing features for Stacks 2.1 upgrade.
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:42]:
```suggestion
Now that you have a general understanding of Stacks 2.1 and the PoX changes and what to expect, let's dive into each product and understand the new features or improvements to the existing features for Stacks 2.1 upgrade.
```

File: docs/stacks-2.1-upgrades.md [hunk 16]
```diff
@@ -80,9 +80,9 @@ Spin up a local Devnet network using the command:
 
 `$ clarinet integrate`
 
-If you have any trouble with the above command, refer to our [troubleshooting guide](clarinet/troubleshooting.md) or report an issue [here](https://github.com/hirosystems/clarinet/issues).
+If you have any trouble with the above command, refer to the [troubleshooting guide](clarinet/troubleshooting.md) or report an issue [here](https://github.com/hirosystems/clarinet/issues).
 
-New stacks-node will spin up. At Bitcoin block height 102, the chainstate will migrate to epoch 2.05, a soft introduced earlier this year, reducing operations costs. At bitcoin block height 106, the chainstate will migrate to epoch 2.1.
+New stacks-node will spin up. At Bitcoin block height 102, the chainstate will migrate to epoch 2.05, a network upgrade introduced introduced earlier this year, reducing operations costs. At bitcoin block height 106, the chainstate will migrate to epoch 2.1.
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:85]:
```suggestion
New stacks-node will spin up. At Bitcoin block height 102, the chainstate will migrate to epoch 2.05, a network upgrade introduced earlier this year, reducing operations costs. At bitcoin block height 106, the chainstate will migrate to epoch 2.1.
```

File: docs/stacks-2.1-upgrades.md [hunk 17]
```diff
@@ -169,7 +169,7 @@ The PoX-2 support is extended to various functions mentioned [below](#new-pox-in
 
 ## Stacks.js
 
-Stacks 2.1 updates for Stacks JS include methods now accepting `poxAddress` in more BTC address formats (P2PKH, P2SH, P2WPKH, P2WSH, P2TR) and new [Stacking](#new-pox-interaction-functions-clarity) and [Helper](#new-helper-methods) methods.
+Stacks 2.1 upgrade for Stacks JS include methods now accepting `poxAddress` in more BTC address formats (P2PKH, P2SH, P2WPKH, P2WSH, P2TR) and new [Stacking](#new-pox-interaction-functions-clarity) and [Helper](#new-helper-methods) methods.
```
Comment (sabbyanandan) [inline docs/stacks-2.1-upgrades.md:172]: Stacks.js is the current naming, so let's retain it.

```suggestion
Stacks 2.1 upgrade for Stacks.js include methods now accepting `poxAddress` in more BTC address formats (P2PKH, P2SH, P2WPKH, P2WSH, P2TR) and new [Stacking](#new-pox-interaction-functions-clarity) and [Helper](#new-helper-methods) methods.
```

**Review by sabbyanandan — CHANGES_REQUESTED** [review]

**Review by trichards-hiro — APPROVED** [review]

**Pull request merged**